include/boost/multiprecision/cpp_bin_float.hpp

```diff
@@ -2154,7 +2154,14 @@ class numeric_limits<boost::multiprecision::number<boost::multiprecision::cpp_bi
    static constexpr bool                                                                                                             has_infinity      = true;
    static constexpr bool                                                                                                             has_quiet_NaN     = true;
    static constexpr bool                                                                                                             has_signaling_NaN = false;
+#ifdef _MSC_VER
+#pragma warning(push)
+#pragma warning(disable:4996)
+#endif
    static constexpr float_denorm_style has_denorm                                                                                                      = denorm_absent;
+#ifdef _MSC_VER
+#pragma warning(pop)
+#endif
    static constexpr bool               has_denorm_loss                                                                                                 = false;
```

Comment:
Do the instances of `has_denorm_loss` need to go in between the pragmas since it's deprecated as well?

Reply:
I don't believe so... it's a simple bool so no warning is generated, it's our use of `float_denorm_style` that was causing the issue I believe.